server/types/interface.go

```diff
@@ -60,6 +60,19 @@ type DoltgresArrayType interface {
 	BaseType() DoltgresType
 }
 
+// DoltgresValidType is a DoltgresType that represents a type displayed in pg_catalog tables.
+type DoltgresValidType interface {
```

Comment:
It looks like every type implements this except for `array` and `null`, so seems like these functions should go in `DoltgresType` instead and we can remove this interface.